### PR TITLE
set config.php to stop working after config is done, refs #14

### PR DESCRIPTION
**Pull request by dissolve**
moved the check for file to the front than then only show setup steps if not configured.

Check for existance of cryptographically secure functions and then warn if not present.

### CODE REVIEW

File: setup.php [hunk 1]
```diff
@@ -75,39 +43,81 @@
         $configured = false;
     }
 
-	$file_written = false;
+?>
 
-    if(is_writeable($configfile) && !$configured){
+<?php if($configured): //TODO, should this reveal exactly how to reconfigure? I don't this it would be a security issue ?>
```
Comment (Zegnat) [inline setup.php]: Telling people to remove config.php if they want to re-run the set-up should not give away any secret information, I think.

File: setup.php [hunk 2]
```diff
@@ -75,39 +43,81 @@
...
+    <h2>System already configured</h2>
+<?php else: ?>
 
-		$handle = fopen($configfile, 'w');
+    <?php if(!function_exists('random_bytes') && !function_exists('openssl_random_pseudo_bytes')): ?>
```
Comment (Zegnat) [inline setup.php]: Should we test for `hash_equals` too?

That is what protects us from timing attacks so while it has no effect on the key generation it is definitely important for how secure running Selfauth down the line is.

Comment (Zegnat) [inline setup.php]: The existence of `openssl_random_pseudo_bytes` is no guarantee that Selfauth is able to generate a cryptographically secure key!

[The function](https://secure.php.net/manual/en/function.openssl-random-pseudo-bytes.php) can set a `$crypto_strong` boolean which is the only way to know whether OpenSSL did its job right. We can’t test for this before the key generation.

**Comment by dissolve** [comment]
@Zegnat updated the PR to merge master and fix these

**Review by Zegnat — APPROVED** [review]
This LGTM, if we start using any other security methods the warning can always be tweaked later. This seems like a great step to inform users.

**Pull request merged**